EU4toV2/Source/V2World/V2World.cpp

```diff
@@ -80,6 +81,212 @@ V2World::V2World(const EU4::world& sourceWorld, const mappers::IdeaEffectMapper&
 	output(potentialGPs);
 }
 
+class BucketList : private commonItems::parser
+{
+      public:
+	explicit BucketList(const std::string& filename);
+	void putInBucket(V2Province* prov);
+	bool empty() const { return buckets.empty(); }
+        void shuffle();
+
+      private:
+	struct Bucket

```

Comment:
Bucket should be extracted into it's own h/cpp files and also made into a class.

Reply:
When extracting both classes into their own files, it may make sense to put them in a subdirectory. Source/V2World/ is pretty cluttered...

Reply:
As these classes are very strongly related I put them in a single source file, which I put in the Helpers directory.